consensus/hotstuff/committees/consensus_committee_test.go

```diff
@@ -31,67 +260,55 @@ func TestConsensus_IdentitiesByBlock(t *testing.T) {
-	snapshot.On("Epochs").Return(epochs)
-
-	state.On("Final").Return(snapshot)
-	state.On("AtBlockID", blockID).Return(snapshot)
+	currEpoch := newMockEpoch(1, unittest.IdentityListFixture(10), 1, 100, unittest.SeedFixture(seed.RandomSourceLength), true)
+	suite.epochs.Add(currEpoch)
 
-	snapshot.On("Identity", realIdentity.NodeID).Return(realIdentity, nil)
-	snapshot.On("Identity", zeroWeightConsensusIdentity.NodeID).Return(zeroWeightConsensusIdentity, nil)
-	snapshot.On("Identity", ejectedConsensusIdentity.NodeID).Return(ejectedConsensusIdentity, nil)
-	snapshot.On("Identity", validNonConsensusIdentity.NodeID).Return(validNonConsensusIdentity, nil)
-	snapshot.On("Identity", fakeID).Return(nil, protocol.IdentityNotFoundError{})
+	suite.state.On("AtBlockID", blockID).Return(suite.snapshot)
+	suite.snapshot.On("Identity", realIdentity.NodeID).Return(realIdentity, nil)
+	suite.snapshot.On("Identity", zeroWeightConsensusIdentity.NodeID).Return(zeroWeightConsensusIdentity, nil)
+	suite.snapshot.On("Identity", ejectedConsensusIdentity.NodeID).Return(ejectedConsensusIdentity, nil)
+	suite.snapshot.On("Identity", validNonConsensusIdentity.NodeID).Return(validNonConsensusIdentity, nil)
+	suite.snapshot.On("Identity", fakeID).Return(nil, protocol.IdentityNotFoundError{})
 
-	com, err := NewConsensusCommittee(state, unittest.IdentifierFixture())
-	require.NoError(t, err)
+	suite.CreateAndStartCommittee()
 
 	t.Run("non-existent identity should return InvalidSignerError", func(t *testing.T) {
-		_, err := com.IdentityByBlock(blockID, fakeID)
+		_, err := suite.committee.IdentityByBlock(blockID, fakeID)
 		require.True(t, model.IsInvalidSignerError(err))
 	})
 
 	t.Run("existent but non-committee-member identity should return InvalidSignerError", func(t *testing.T) {
 		t.Run("zero-weight consensus node", func(t *testing.T) {
-			_, err := com.IdentityByBlock(blockID, zeroWeightConsensusIdentity.NodeID)
+			_, err := suite.committee.IdentityByBlock(blockID, zeroWeightConsensusIdentity.NodeID)
 			require.True(t, model.IsInvalidSignerError(err))
 		})
 
 		t.Run("ejected consensus node", func(t *testing.T) {
-			_, err := com.IdentityByBlock(blockID, ejectedConsensusIdentity.NodeID)
+			_, err := suite.committee.IdentityByBlock(blockID, ejectedConsensusIdentity.NodeID)
 			require.True(t, model.IsInvalidSignerError(err))
 		})
 
 		t.Run("otherwise valid non-consensus node", func(t *testing.T) {
-			_, err := com.IdentityByBlock(blockID, validNonConsensusIdentity.NodeID)
+			_, err := suite.committee.IdentityByBlock(blockID, validNonConsensusIdentity.NodeID)
 			require.True(t, model.IsInvalidSignerError(err))
 		})
 	})
 
 	t.Run("should be able to retrieve real identity", func(t *testing.T) {
-		actual, err := com.IdentityByBlock(blockID, realIdentity.NodeID)
+		actual, err := suite.committee.IdentityByBlock(blockID, realIdentity.NodeID)
 		require.NoError(t, err)
 		require.Equal(t, realIdentity, actual)
 	})
 }
```

Comment:
I suspect that this test case would pass if `committee.Consensus` wrapped _all_ errors  returned by the `protocol.State` [here](https://github.com/onflow/flow-go/blob/b1a97ec92c979fbf31630552bb2e805e316479d2/consensus/hotstuff/committees/consensus_committee.go#L189-L190) as `InvalidSignerError` _irrespective of their type_. 

But this would not be correct. If `protocol.State` returns anything else than `IdentityNotFound`, the exception should be bubbled up the call stack and _not_ interpreted as "identity not found". I think it would be worth while testing this as well.

Reply:
https://github.com/onflow/flow-go/pull/2782/commits/5ee8505fe0e73bcce7ea0b9409d54ec5c92729a7